addons/mail/static/src/components/message/message.scss

```diff
@@ -106,6 +106,12 @@
     font-size: 0.8em;
 }
 
+.o_Message_highlightIndicator {
+    width: 4px;
+    margin-top: -0.5rem;
+    margin-bottom: -0.5rem;
```

Comment:
Avoid negative margin, remove the padding from the parent element and add margin on children element to compensate.